R/epidist_utils.R

```diff
@@ -0,0 +1,605 @@
+                                         sd_ci_interval = NA_real_,
+                                         median = NA_real_,
+                                         median_ci = c(NA_real_, NA_real_),
+                                         median_ci_interval = NA_real_,
+                                         dispersion = NA_real_,
+                                         dispersion_ci = c(NA_real_, NA_real_),
+                                         dispersion_ci_interval = NA_real_,
+                                         lower_range = NA_real_,
+                                         upper_range = NA_real_,
+                                         q_025 = NA_real_,
+                                         q_05 = NA_real_,
+                                         q_25 = NA_real_,
+                                         q_75 = NA_real_,
+                                         q_875 = NA_real_,
+                                         q_95 = NA_real_,
+                                         q_975 = NA_real_) {
+
+  # check input
+  checkmate::assert_number(mean, na.ok = TRUE)
+  checkmate::assert_numeric(mean_ci, len = 2, any.missing = TRUE)
+  checkmate::assert_number(mean_ci_interval, na.ok = TRUE)
+  checkmate::assert_number(sd, na.ok = TRUE)
+  checkmate::assert_numeric(sd_ci, len = 2, any.missing = TRUE)
+  checkmate::assert_number(sd_ci_interval, na.ok = TRUE)
+  checkmate::assert_number(median, na.ok = TRUE)
+  checkmate::assert_numeric(median_ci, len = 2, any.missing = TRUE)
+  checkmate::assert_number(median_ci_interval, na.ok = TRUE)
+  checkmate::assert_number(dispersion, na.ok = TRUE)
+  checkmate::assert_numeric(dispersion_ci, len = 2, any.missing = TRUE)
+  checkmate::assert_number(dispersion_ci_interval, na.ok = TRUE)
+  checkmate::assert_number(lower_range, na.ok = TRUE)
+  checkmate::assert_number(upper_range, na.ok = TRUE)
+  checkmate::assert_number(q_025, na.ok = TRUE)
+  checkmate::assert_number(q_05, na.ok = TRUE)
+  checkmate::assert_number(q_25, na.ok = TRUE)
+  checkmate::assert_number(q_75, na.ok = TRUE)
+  checkmate::assert_number(q_95, na.ok = TRUE)
+  checkmate::assert_number(q_975, na.ok = TRUE)
+
+  # return list of summary stats
+  list(
+    central_tendency_spread = list(
+      mean = mean,
+      mean_ci = mean_ci,
+      mean_ci_interval = mean_ci_interval,
+      sd = sd,
+      sd_ci = sd_ci,
+      sd_ci_interval = sd_ci_interval,
+      median = median,
+      median_ci = median_ci,
+      median_ci_interval = median_ci_interval
+    ),
+    quantiles = list(
+      q_025 = q_025,
+      q_05 = q_05,
+      q_25 = q_25,
+      q_50 = median,
+      q_75 = q_75,
+      q_875 = q_875,
+      q_95 = q_95,
+      q_975 = q_975
+    ),
+    range = list(lower_range = lower_range, upper_range = upper_range),
+    dispersion = list(
+      dispersion = dispersion,
+      dispersion_ci = dispersion_ci,
+      dispersion_ci_interval = dispersion_ci_interval
+    )
+  )
+}
```

Comment:
1. Suggest changing function title to not be the same as the description, suggest instead "Create epidist summary statistics".
2. Suggest renaming all "statistic_ci" to follow naming of `create_epidist_uncertainty()`: "ci" for the interval width, and "ci_limits" for the lower and upper limits.
3. Suggest adding a realistic working example.

Reply:
Updated documentation (title and examples) for `create_epidist_summary_stats()` in a1ba591. Naming of ci fields was updated in a previous commit ad66720